static/app/components/events/interfaces/spans/index.tsx

```diff
@@ -37,14 +34,14 @@ type Props = {
 type State = {
   parsedTrace: ParsedTraceType;
   searchQuery: string | undefined;
-  operationNameFilters: ActiveOperationFilter;
+  waterfallModel: WaterfallModel;
 };
 
-class SpansInterface extends Component<Props, State> {
+class SpansInterface extends PureComponent<Props, State> {
```

Comment:
How well does `PureComponent` play with `WaterfallModel` here? I thought `PureComponents` don't update/rerender when an object in the state gets mutated?

Reply:
Did some further reading, looks like this is all handled by `mobx`. `WaterfallModel` is made `observable` and the `Observer` component triggers the rerender whenever `WaterfallModel` attributes in use are updated.

Reply:
@edwardgou-sentry  For this component, the `PureComponent` change doesn't interact with `WaterfallModel` in any way. 

This was a change I brought over from a git branch containing most of the refactor code. I don't think this one-line change would be worth a separate pull-request 😬 . Unless you strongly feel it does 😄 

-----

The higher-order component wrapper (`observable` -- lower-case o) isn't used here. And thus, `shouldComponentUpdate()` isn't provided by mobx.

The `<Observer />` components used here are component "regions" that are re-rendered (the render-prop function is called) whenever an observed mobx state changes.  The rest of the `render()` contents remains unchanged.

Reply:
👍  makes sense to me! Let's keep the `PureComponent` change, don't see any issues with that.